### DISCUSSION TITLE
Update to cmake 3.28

### DISCUSSION THREAD

**Discussion by tigrux** [facebookincubator/velox#9792] [Ideas]
CMake 3.24 and the 3.28 provide two improvements to the FetchContent module:
https://cmake.org/cmake/help/v3.28/module/FetchContent.html

New in version 3.24: OVERRIDE_FIND_PACKAGE - This makes the content available to be available for find_package.
New in version 3.28: EXCLUDE_FROM_ALL - Prevents the targets of the content from being added to the all target.

I created a PR that uses cmakes 3.28.3 to simplify the respective .cmake of protobuf, and also adds the dependencies of Google Cloud C++ (Storage and BigQuery) including Abseil and gRPC:
https://github.com/facebookincubator/velox/pull/9072/

The PR is provided as a WIP and I think it may be divided into 3 PRs:
- Update to cmake 3.28.
- Refactor the existing cmake modules.
- Add the dependencies of Google Cloud CPP as cmake modules.

**Comment by tigrux** [comment]
Hello @kgpai, @pedroerp, @majetideepak, @assignUser, @bdice.
As suggested, this is the discussion to consider the upgrade to cmake 3.28.

**Comment by assignUser** [comment]
> New in version 3.24: OVERRIDE_FIND_PACKAGE - This makes the content available to be available for find_package.

This is really awesome and makes it so much easier to add new packages. CMake intercepts the `find_package` call from sub projects and forces it to look in a specific place where `fetchContent` puts a config file. This would otherwise only be possible by patching the calls in the other project.

I am for the bump, cmake is really easy to install in user space and the more recent versions come with great new features like the ones mentioned or file sets (which make installing things MUCH easier). I don't really see a reason to stick with an older version.

**Comment by kgpai** [comment]
Hi Sandino, I think it would be great to add the new features  3.28 - If can ensure that we update all the images and see no problems on the supported platforms, I am all for it

**Comment by tigrux** [comment]
I removed the WIP mark from the PR.

**Comment by tigrux** [comment]
I created a PR with changes specific to the setup scripts:
https://github.com/facebookincubator/velox/pull/9861

**Comment by tigrux** [comment]
The PR was updated and approved, just waiting for its merge.
https://github.com/facebookincubator/velox/pull/9861

**Comment by tigrux** [comment]
Closing discussion, the PR was merged.
